sync/sync.go

```diff
@@ -199,6 +208,7 @@ func (s *Synchronizer) syncBlocks(syncCtx context.Context) {
 				streamCancel()
 				fetchers.Wait()
 				verifiers.Wait()
+				pendingSem <- struct{}{}
```

Comment:
I don't understand why we are sending pendingSem when we streamCtx is done

Reply:
this https://github.com/NethermindEth/juno/pull/780#discussion_r1227946183

to ensure there is no pending poll routine in flight